test/NuGet.Core.Tests/NuGet.Configuration.Test/SearchTreeTest.cs

```diff
@@ -222,9 +222,50 @@ public void SearchTree_InvalidSearchInput_Throws(string packagePatterns, string
             Assert.Equal(expectedLine, exception.Message);
         }
 
+        /// <summary>
+        /// SearchPatternByTerm
+        /// </summary>
+        /// <param name="packagePatterns"></param>
+        /// <returns></returns>
+        [Theory]
+        [InlineData("public,Nuget", "Nuget", "nuget")]
+        [InlineData("public,nuget", "Nuget", "nuget")]
+        [InlineData("public,Nuget", "nuget", "nuget")]
+        [InlineData("public,nuget", "nuget", "nuget")]
+        [InlineData("public,nuget", " nuget", "nuget")]
+        [InlineData("public,nuget", "nuget ", "nuget")]
+        [InlineData("public,nuget", " nuget ", "nuget")]
+        [InlineData("public, nuget", "nuget", "nuget")]
+        [InlineData("public,nuget ", "nuget", "nuget")]
+        [InlineData("public, nuget ", "nuget", "nuget")]
+        [InlineData("public, nuget ", " nuget ", "nuget")]
+        [InlineData(" public , nuget ", " nuget ", "nuget")]
+        [InlineData("   public    ,    nuget    ", "   nuget   ", "nuget")]
+        [InlineData("public,*", "Microsoft.Build", "*")]
+        [InlineData("public,Contoso.Opensource.*,*", "Microsoft.Build", "*")]
+        [InlineData("public,Contoso.Opensource.*", "Contoso.Opensource.", "contoso.opensource.*")]
+        [InlineData("public,Contoso.Opensource.*", "Contoso.Opensource.MVC", "contoso.opensource.*")]
+        [InlineData("public,Contoso.Opensource.*", "Contoso.Opensource.MVC.ASP", "contoso.opensource.*")]
+        [InlineData("public,Contoso.Opensource.* ", "Contoso.Opensource.MVC.ASP", "contoso.opensource.*")]
+        [InlineData(" public,Contoso.Opensource.*", "Contoso.Opensource.MVC.ASP", "contoso.opensource.*")]
+        [InlineData(" public,Contoso.Opensource.* ", " Contoso.Opensource.MVC.ASP ", "contoso.opensource.*")]
+        public void SearchPatternByTerm_ExistingTerm_PatternFound(string packagePatterns, string term, string expectedPatternMatch)
```

Comment:
Notice expected patterns are all lowercase since `SearchTree` lowercases them. Since these should be case insensitive, I think this is sufficient. The UI still renders the original casing.